src/scripts/eastern_kingdoms/tirisfal_glades/scarlet_monastery/boss_mograine_and_whitemane.cpp

```diff
@@ -84,7 +84,8 @@ struct boss_scarlet_commander_mograineAI : public ScriptedAI
         m_uiHammerOfJustice_Timer = 15000;
 
         //Incase wipe during phase that mograine fake death
-        m_creature->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NOT_SELECTABLE | UNIT_FLAG_SPAWNING);
+        m_creature->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_SPAWNING);
```

Comment:
Spawning flag should not be used in script at all. Previously it was named "non attackable" so a lot of scripts used it when they shouldn't have. Can you refactor the script to not use it? Combining Immune to Player and NPC together should have the same effect (non attackable by all).

Reply:
Ok, I will try.

Reply:
I had to add some more changes. The Problem was that Mograine is in combat when he gets the flags Immune to Player and NPC added, and so he would immediately leave fake death and attack player during next `m_creature->SelectHostileTarget() ` check. Now it should work like expected (I did a few tests with bots).